internal/config/config.go

```diff
@@ -1,144 +1,126 @@
+	Kubelet      `mapstructure:"kubelet"`
+	KSM          `mapstructure:"ksm"`
 }
 
 type KSM struct {
-	// URL defines a static endpoint for KSM.
-	StaticURL string
-
-	// Autodiscovery settings.
-	// Scheme that will be used to hit the endpoints of discovered KSM services. Defaults to http.
-	Scheme string
-	// If set, Port will discard all endpoints discovered that do not use this specified port. Otherwise, all endpoints will be considered.
-	Port int
-	// PodLabel is the selector used to filter Endpoints.
-	PodLabel string
-	// Namespace can be used to restric the search to a particular namespace.
-	Namespace string
-	// If set, Distributed will instruct the integration to scrape all KSM endpoints rather than just the first one.
-	Distributed bool
-
-	Enabled bool
+	StaticURL   string `mapstructure:"staticURL"`
+	Scheme      string `mapstructure:"scheme"`
+	Port        int    `mapstructure:"port"`
+	Selector    string `mapstructure:"selector"`
+	Namespace   string `mapstructure:"namespace"`
+	Distributed bool   `mapstructure:"distributed"`
+	Enabled     bool   `mapstructure:"enabled"`
+}
+
+type Kubelet struct {
+	Enabled bool   `mapstructure:"enabled"`
+	Port    int32  `mapstructure:"port"`
+	Scheme  string `mapstructure:"scheme"`
 }
 
 type ControlPlane struct {
-	ETCD              ETCD
-	APIServer         APIServer
-	Scheduler         Scheduler
-	ControllerManager ControllerManager
-	Enabled           bool
+	Enabled           bool              `mapstructure:"enabled"`
+	ETCD              ETCD              `mapstructure:"etcd"`
+	APIServer         APIServer         `mapstructure:"apiServer"`
+	ControllerManager ControllerManager `mapstructure:"controllerManager"`
+	Scheduler         Scheduler         `mapstructure:"scheduler"`
+}
+
+type ETCD struct {
+	Enabled        bool                       `mapstructure:"enabled"`
+	StaticEndpoint StaticEndpoint             `mapstructure:"staticEndpoint"`
+	Autodiscover   []AutodiscoverControlPlane `mapstructure:"autodiscover"`
 }
 
 type APIServer struct {
-	APIServerEndpointURL string
-	APIServerSecurePort  string
+	Enabled        bool                       `mapstructure:"enabled"`
+	StaticEndpoint StaticEndpoint             `mapstructure:"staticEndpoint"`
+	Autodiscover   []AutodiscoverControlPlane `mapstructure:"autodiscover"`
 }
 
-type ETCD struct {
-	EtcdEndpointURL        string
-	EtcdTLSSecretNamespace string
-	EtcdTLSSecretName      string
+type ControllerManager struct {
+	Enabled        bool                       `mapstructure:"enabled"`
+	StaticEndpoint StaticEndpoint             `mapstructure:"staticEndpoint"`
+	Autodiscover   []AutodiscoverControlPlane `mapstructure:"autodiscover"`
 }
 
 type Scheduler struct {
-	SchedulerEndpointURL string
+	Enabled        bool                       `mapstructure:"enabled"`
+	StaticEndpoint StaticEndpoint             `mapstructure:"staticEndpoint"`
+	Autodiscover   []AutodiscoverControlPlane `mapstructure:"autodiscover"`
```

Comment:
would make sense to create a controlplane config abstraction and load a map of that? instead of having a struct per each component just have a single one i mean.